torchaudio/datasets/librispeech.py

```diff
@@ -43,16 +43,16 @@ def download_librispeech(root, url):
     extract_archive(archive)
 
 
-def load_librispeech_item(
+def get_librispeech_metadata(
```

Comment:
Please move this to class method so that it's overridable from client code.

Reply:
Same goes to `load_librispeech_item`.

Reply:
is this preferable to having users directly override the `__getitem__` method of the dataset (as opposed to overriding a class method that `__getitem__` calls which involves going a layer deeper)? also since `load_librispeech_item` is currently being used by another dataset/accessible to users

Reply:
`load_librispeech_item` should not be public. Can we follow up and make it private?

From the regular OOP perspective, the right approach is to let users overwrite `__getitem__`. However, this forces the users to re-write/copy-paste almost the whole logic of path-parsing and text processing. There are situations where only decoding should be overwritten but since these functions being module-level plain function, it is not plausible. These functions being class method allow to override the logic in the subclass. That's the pattern we should have been followed in the dataset class implementations, but existing datasets not doing that led to the situation where new dataset did not follow either. See https://github.com/pytorch/audio/issues/910#issuecomment-693564762 for the basic idea.